src/Compilers/CSharp/Test/Syntax/Syntax/FieldAndValueKeywordTests.cs

```diff
@@ -0,0 +1,1371 @@
+                            object F1(object field) => null;
+                            object F2(object @field) => null;
+                            return null;
+                        }
+                        set
+                        {
+                            void G1(object value) { }
+                            void G2(object @value) { }
+                        }
+                    }
+                }
+                """;
+            var comp = CreateCompilation(source, parseOptions: TestOptions.Regular12);
+            comp.VerifyEmitDiagnostics(
+                // (8,23): info CS9248: 'field' is a contextual keyword, with a specific meaning, starting in language version preview. Use '@field' to avoid a breaking change when compiling with language version preview or later.
+                //             object F1(object field) => null;
+                Diagnostic(ErrorCode.INF_IdentifierConflictWithContextualKeyword, "object field").WithArguments("field", "preview").WithLocation(8, 23),
+                // (14,21): info CS9248: 'value' is a contextual keyword, with a specific meaning, starting in language version preview. Use '@value' to avoid a breaking change when compiling with language version preview or later.
+                //             void G1(object value) { }
+                Diagnostic(ErrorCode.INF_IdentifierConflictWithContextualKeyword, "object value").WithArguments("value", "preview").WithLocation(14, 21));
+        }
+
+        [Theory]
+        [CombinatorialData]
+        public void Attribute_01(
+            [CombinatorialValues(LanguageVersion.CSharp12, LanguageVersion.Preview)] LanguageVersion languageVersion, bool escapeIdentifier)
+        {
+            string identifier = escapeIdentifier ? "@field" : "field";
+            string source = $$"""
+                using System;
+                class A : Attribute
+                {
+                    public A(string s) { }
+                }
+                class C
+                {
+                    const int field = 0;
+                    [A(nameof({{identifier}}))]
+                    object P
+                    {
+                        [A(nameof({{identifier}}))] get { return null; }
+                        [A(nameof({{identifier}}))] set { }
+                    }
+                    [A(nameof({{identifier}}))]
+                    event EventHandler E
+                    {
+                        [A(nameof({{identifier}}))] add { }
+                        [A(nameof({{identifier}}))] remove { }
+                    }
+                }
+                """;
+            var comp = CreateCompilation(source, parseOptions: TestOptions.Regular.WithLanguageVersion(languageVersion));
+            if (languageVersion > LanguageVersion.CSharp12 || escapeIdentifier)
+            {
+                comp.VerifyEmitDiagnostics();
+            }
+            else
+            {
+                comp.VerifyEmitDiagnostics(
```

Comment:
>VerifyEmitDiagnostics [](http://example.com/codeflow?start=21&length=21)

The diagnostics here are incorrect. No diagnostics should be reported because `[A(nameof(field))]` is not within the scope of the accessor.